pygen/pygen_src/riscv_instr_cover_group.py

```diff
@@ -0,0 +1,83 @@
+"""Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import vsc
+from pygen.pygen_src.riscv_instr_pkg import *
+from pygen.pygen_src.isa.riscv_cov_instr import *
+
+
+class riscv_instr_cover_group:
+    def __init__(self):
+        self.pre_instr = riscv_cov_instr()
+        self.cfg = None
+        self.instr_list = []
+        self.instr_cnt = 0
+        self.branch_instr_cnt = 0
+        self.branch_hit_history = vsc.bit_t(5)  # The last 5 branch result
+        self.ignored_exceptions = []
+        self.exception_list = []
+        ''' 
+        Mode of the coverage model:
+        In complicance mode, all the micro-architecture related covergroups 
+        are removed. Only the ones related to RISC-V specification compliance 
+        is sampled.
+        '''
+        self.compliance_mode = vsc.bit_t(1)
+        self.select_isa = vsc.bit_t(1)  # Select an ISA extension to cover
+        self.cov_isa = None
+
+    @vsc.covergroup
+    class r_instr_cg(object):

```

Comment:
general question: does pyvsc coverage have an equivalent to `covergroup cg with function sample(...)` in SV?

Reply:
No, unfortunately pyvsc doesn't have an equivalent to the `covergroup X with function sample(input)`. Although it makes the code much cleaner & easier to read, I don't think that this is an essential feature, right?

Reply:
Yes, generally this feature isn't absolutely essential - since there's only one sample() function being used here it's fine, I was just wondering if this was supported.